### PR TITLE
checker, statistic: avoid leak in label statistic

### LINKED ISSUE #8700: `labelLevelStats` may leak when merging regions
## Bug Report

Consider the following case:

1. Patrol region has scanned a batch of regions, e.g., a,b,c,d
2. Before updating label statistics, the merge finished, e.g., merge a and b
3. PD will remove one of the regions, assumed b, from label statistics because of overlaps during the process of region heartbeat.
4. Then patrol region update a,b,c,d to label statistics

b will never be deleted again and left in memory until the PD is restarted.

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #8700

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/checker/checker_controller.go [hunk 1]
```diff
@@ -141,6 +141,8 @@ func (c *Controller) PatrolRegions() {
 			c.cluster.UpdateRegionsLabelLevelStats(regions)
 			// When the key is nil, it means that the scan is finished.
 			if len(key) == 0 {
+				// Clear the defunct regions label level statistics.
+				c.cluster.ClearDefunctRegionsLabelLevelStats()
```
[the remark this answers is not in the document]
Comment (lhy1024) [reply pkg/schedule/checker/checker_controller.go:145]: No, only clear defunct regions after scanning a round finish.

[the remark this answers is not in the document]
Comment (lhy1024) [reply pkg/schedule/checker/checker_controller.go:145]: I got

File: pkg/statistics/region_collection.go [hunk 2]
```diff
@@ -405,14 +407,26 @@ func ResetLabelStatsMetrics() {
...
 	defer l.Unlock()
-	if label, ok := l.regionLabelStats[regionID]; ok {
-		l.labelCounter[label]--
-		delete(l.regionLabelStats, regionID)
+	l.defunctRegions[regionID] = struct{}{}
+}
+
+// ClearDefunctRegions is used to handle the overlap region.
+// It is used to remove the defunct regions from the label statistics.
+func (l *LabelStatistics) ClearDefunctRegions() {
```
Comment (rleungx) [inline pkg/statistics/region_collection.go:420]: How about adding a unit test?

Comment (lhy1024) [reply pkg/statistics/region_collection.go:420]: ok, I have added a unit test, which can pass in this pr and is failed in master branch.

**Review by rleungx — APPROVED** [review]

**Pull request merged**